dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java

```diff
@@ -428,9 +431,27 @@ public boolean isSuccessOfYarnState(List<String> appIds) {
             result = false;
         }
         return result;
-
     }
-
+    
+    /**
+     *
+     * @param appId
+     * @return appId ExecutionStatus
+     * @throws Exception
+     */
+    private ExecutionStatus retryExecutionStatus(String appId,ExecutionStatus applicationStatus) throws Exception {
+        for (int i = 0; i < 10; i++) {
+            applicationStatus = HadoopUtils.getInstance().getApplicationStatus(appId);
+            if (applicationStatus.equals(ExecutionStatus.FAILURE)
+                    || applicationStatus.equals(ExecutionStatus.KILL)) {
```

Comment:
Why do  need to retry in failure and kill state?

Reply:
because the spark or flink application on yarn can Automatic fault tolerance！maybe they can start up again，so give them time attempt

Reply:
![image](https://user-images.githubusercontent.com/9714395/132296305-661d6c6b-c722-4089-87a3-d690f38eb246.png)
just like this!
then retry 10 time Still failure，so it failed